ethereum/contracts/pyth/PythGovernance.sol

```diff
@@ -0,0 +1,118 @@
+// contracts/Governance.sol
+// SPDX-License-Identifier: Apache 2
+
+pragma solidity ^0.8.0;
+
+import "./PythGovernanceInstructions.sol";
+import "./PythInternalStructs.sol";
+import "./PythGetters.sol";
+import "./PythSetters.sol";
+
+
+import "@openzeppelin/contracts/proxy/ERC1967/ERC1967Upgrade.sol";
+
+/**
+ * @dev `Governance` defines a means to enacting changes to the Pyth contract.
+ */
+abstract contract PythGovernance is PythGetters, PythSetters, PythGovernanceInstructions {
+    event ContractUpgraded(address oldContract, address newContract);
+    event GovernanceDataSourceSet(PythInternalStructs.DataSource oldDataSource, PythInternalStructs.DataSource newDataSource,
+        uint64 initialSequence);
+    event DataSourcesSet(PythInternalStructs.DataSource[] oldDataSources, PythInternalStructs.DataSource[] newDataSources);
+    event FeeSet(uint oldFee, uint newFee);
+    event ValidPeriodSet(uint oldValidPeriod, uint newValidPeriod);
+
+    function verifyGovernanceVM(bytes memory encodedVM) internal returns (IWormhole.VM memory parsedVM){
+        (IWormhole.VM memory vm, bool valid, string memory reason) = wormhole().parseAndVerifyVM(encodedVM);
+
+        require(valid, reason);
+        require(isValidGovernanceDataSource(vm.emitterChainId, vm.emitterAddress), "VAA is not coming from the governance data source");
+
+        require(vm.sequence > lastExecutedGovernanceSequence(), "VAA is older than the last executed governance VAA");
+
+        setLastExecutedGovernanceSequence(vm.sequence);
+
```

Comment:
I suggest adding a magic number into the VAA that we can use to verify that this message is intended to be interpreted as a `GovernanceInstruction`. I'm a little worried that someone could trick the emitter to sign a VAA payload that is simultaneously valid for multiple different uses. I think that's pretty unlikely, but a magic number would make it very clear what the possible uses for any given VAA payload are.

Reply:
(e.g., pyth price update VAAs all start with the byte sequence "P2WH" as their magic number)